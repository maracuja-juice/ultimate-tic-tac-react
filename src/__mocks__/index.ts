import {
  unfinishedBoardMock,
  movesForUnfinishedBoardMock,
} from './unfinishedBoardMock';
import {
  crossFinishedGameMock,
  circleFinishedGameMock,
} from './finishedGameDataMock';
import {
  circleFinishedBoardMock,
  crossFinishedBoardMock,
  movesForCircleFinishedBoardMock,
  movesForCrossFinishedBoardMock,
} from './finishedBoardMock';
import {
  activeBoardsForBoardWithTwoMovesMock,
  boardWithTwoMovesMock,
} from './boardWithTwoMovesMock';
import {
  boardWithThreeMovesMock,
  activeBoardsForBoardWithThreeMovesMock,
  movesForBoardWithThreeMovesMock,
} from './boardWithThreeMovesMock';
import {
  boardWithOneMoveMock,
  activeBoardsForBoardWithOneMoveMock,
} from './boardWithOneMoveMock';

export {
  unfinishedBoardMock,
  movesForUnfinishedBoardMock,
  crossFinishedGameMock,
  circleFinishedGameMock,
  circleFinishedBoardMock,
  movesForCircleFinishedBoardMock,
  crossFinishedBoardMock,
  movesForCrossFinishedBoardMock,
  boardWithTwoMovesMock,
  activeBoardsForBoardWithTwoMovesMock,
  boardWithThreeMovesMock,
  movesForBoardWithThreeMovesMock,
  activeBoardsForBoardWithThreeMovesMock,
  boardWithOneMoveMock,
  activeBoardsForBoardWithOneMoveMock,
};
