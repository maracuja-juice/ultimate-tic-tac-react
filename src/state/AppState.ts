import { Point } from '../util';

export interface AppState {
  currentGame: GameInformation;
  finishedGames: FinishedGameState[];
  analysisGame: AnalysisGame;
}

export interface GameInformation {
  board: SmallBoardInformation[];
  game: GameState;
  moves: MoveState[];
  activeBoards: Point[];
}

export interface FinishedGameState {
  id?: string;
  date: string;
  winner: Winner;
  gameState: SmallBoardInformation[];
  moves: MoveState[];
  saveState: string;
  errorMessage: string;
}

export interface AnalysisGame {
  id: string;
  board: SmallBoardInformation[];
  game: GameState;
  moves: MoveState[];
  activeBoards: Point[];
  currentMove: number;
}

export interface GameState {
  currentPlayer: Player;
  isFinished: boolean;
  winningPlayer: Winner;
}

export interface TileInformation {
  position: Point;
  value: TileValue;
}

export interface SmallBoardInformation extends TileInformation {
  tiles: SmallTileInformation[];
}

export interface SmallTileInformation extends TileInformation {
  boardPosition: Point;
}

export enum TileValue {
  Cross = 0,
  Circle = 1,
  Empty = 2,
  Destroyed = 3,
}

export enum Winner {
  Cross = 0,
  Circle = 1,
  None = 2,
  Draw = 3,
}

export enum Player {
  Cross = 0,
  Circle = 1,
}

export interface Move {
  boardPosition: Point;
  tilePosition: Point;
}

export interface MoveState extends Move {
  player: Player;
  moveNumber: number;
}

export interface GenericAction {
  type: string;
  // tslint:disable-next-line: no-any
  payload?: any;
}

export interface MarkSpecially {
  condition: boolean;
  position?: { boardPosition: Point; tilePosition: Point };
}
