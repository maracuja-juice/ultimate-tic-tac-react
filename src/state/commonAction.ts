export const RESTART_GAME = 'common/restart-game';

export const restartGame = () => ({
    type: RESTART_GAME
});