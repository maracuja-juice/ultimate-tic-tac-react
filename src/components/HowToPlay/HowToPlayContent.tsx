import * as React from 'react';
import { HowToPlayBoardState } from '../../state/AppState';
import { BigBoard } from '../Board/BigBoard/BigBoard';

interface HowToPlayContentProps {
  boardState: HowToPlayBoardState;
}

export function HowToPlayContent(props: HowToPlayContentProps) {
  const { boardState } = props;
  return (
    <BigBoard
      animate={boardState.animate}
      currentPlayer={boardState.currentPlayer}
      board={boardState.board}
      activeBoards={boardState.activeBoards}
    />
  );
}
