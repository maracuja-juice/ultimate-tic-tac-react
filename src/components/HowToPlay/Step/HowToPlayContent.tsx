import * as React from 'react';
import { BigBoard } from '../../Board/BigBoard/BigBoard';
import { HowToPlayBoardState } from '../../../state/AppState';

interface HowToPlayContentProps {
  boardState: HowToPlayBoardState;
}

export function HowToPlayContent(props: HowToPlayContentProps) {
  const { boardState } = props;
  return (
    <BigBoard
      animate={boardState.animate}
      currentPlayer={boardState.currentPlayer}
      board={boardState.board}
      activeBoards={boardState.activeBoards}
    />
  );
}
