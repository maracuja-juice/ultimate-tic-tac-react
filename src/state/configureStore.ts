import { AppState } from './AppState';
import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import tileReducer from './tiles/tileReducer';
import { composeWithDevTools } from 'redux-devtools-extension';
import gameReducer from './game/gameReducer';
import moveReducer from './moves/moveReducer';
import { enableBatching } from 'redux-batched-actions';
import activeBoardReducer from './activeBoards/activeBoardsReducer';
import createSagaMiddleware from 'redux-saga';
import { all, fork } from 'redux-saga/effects';
import moveValidationSaga from './sagas/MoveValidationSaga';

const rootreducer = combineReducers<AppState>(
    {
        game: gameReducer,
        tiles: tileReducer,
        moves: moveReducer,
        activeBoards: activeBoardReducer
    } );

export function configureStore() {

    const sagaMiddleware = createSagaMiddleware();

    const middleWaresToApply = [
        logger,
        sagaMiddleware
    ];
    const middleware = applyMiddleware( ...middleWaresToApply );
    const store = createStore( enableBatching( rootreducer ), composeWithDevTools(
        middleware
    ) );

    sagaMiddleware.run( rootSaga );

    return store;
}

function* rootSaga() {
    yield all(
        [
            fork( moveValidationSaga ),
        ]
    );
}