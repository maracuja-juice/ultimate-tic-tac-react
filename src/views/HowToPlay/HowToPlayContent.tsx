import * as React from 'react';
import { HowToPlayBoardState } from '../../state/AppState';
import { BigBoard } from '../../components/Board/BigBoard/BigBoard';

interface HowToPlayContentProps {
  boardState: HowToPlayBoardState;
}

export function HowToPlayContent({ boardState }: HowToPlayContentProps) {
  return (
    <BigBoard
      animate={boardState.animate}
      currentPlayer={boardState.currentPlayer}
      board={boardState.board}
      activeBoards={boardState.activeBoards}
    />
  );
}
