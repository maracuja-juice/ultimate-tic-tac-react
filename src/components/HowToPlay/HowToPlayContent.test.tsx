import * as React from 'react';
import { shallow } from 'enzyme';
import { Player } from '../../state/AppState';
import { unfinishedBoardMock } from '../../__mocks__';
import { HowToPlayContent } from './HowToPlayContent';

describe('HowToPlayContent', () => {
  it('should match snapshot', () => {
    const howToPlayContent = shallow(
      <HowToPlayContent
        boardState={{
          board: unfinishedBoardMock,
          activeBoards: [{ x: 0, y: 0 }],
          animate: true,
          currentPlayer: Player.Cross,
        }}
      />,
    );

    expect(howToPlayContent).toMatchSnapshot();
  });
});
