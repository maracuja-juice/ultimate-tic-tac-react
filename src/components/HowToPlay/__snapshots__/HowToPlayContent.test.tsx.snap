// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToPlayContent should match snapshot 1`] = `
<BigBoard
  activeBoards={
    Array [
      Object {
        "x": 0,
        "y": 0,
      },
    ]
  }
  animate={true}
  board={
    Array [
      Object {
        "position": Object {
          "x": 0,
          "y": 0,
        },
        "tiles": Array [
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 0,
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 0,
            },
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 0,
            },
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 0,
            },
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 0,
            },
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 0,
            },
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 0,
            },
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 0,
            },
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 0,
            },
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "value": 2,
          },
        ],
        "value": 0,
      },
      Object {
        "position": Object {
          "x": 0,
          "y": 1,
        },
        "tiles": Array [
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 1,
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 1,
            },
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 1,
            },
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 1,
            },
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 1,
            },
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 1,
            },
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 1,
            },
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 1,
            },
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 1,
            },
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "value": 2,
          },
        ],
        "value": 0,
      },
      Object {
        "position": Object {
          "x": 0,
          "y": 2,
        },
        "tiles": Array [
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 2,
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 2,
            },
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 2,
            },
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 2,
            },
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 2,
            },
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 2,
            },
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 2,
            },
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 2,
            },
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 0,
              "y": 2,
            },
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "value": 2,
          },
        ],
        "value": 2,
      },
      Object {
        "position": Object {
          "x": 1,
          "y": 0,
        },
        "tiles": Array [
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 0,
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 0,
            },
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 0,
            },
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 0,
            },
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 0,
            },
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 0,
            },
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 0,
            },
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 0,
            },
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 0,
            },
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "value": 2,
          },
        ],
        "value": 2,
      },
      Object {
        "position": Object {
          "x": 1,
          "y": 1,
        },
        "tiles": Array [
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 1,
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 1,
            },
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 1,
            },
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 1,
            },
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 1,
            },
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 1,
            },
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 1,
            },
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 1,
            },
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 1,
            },
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "value": 2,
          },
        ],
        "value": 1,
      },
      Object {
        "position": Object {
          "x": 1,
          "y": 2,
        },
        "tiles": Array [
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 2,
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 2,
            },
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 2,
            },
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 2,
            },
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 2,
            },
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 2,
            },
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 2,
            },
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 2,
            },
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 1,
              "y": 2,
            },
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "value": 2,
          },
        ],
        "value": 2,
      },
      Object {
        "position": Object {
          "x": 2,
          "y": 0,
        },
        "tiles": Array [
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 0,
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 0,
            },
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 0,
            },
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 0,
            },
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 0,
            },
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 0,
            },
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 0,
            },
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 0,
            },
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 0,
            },
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "value": 2,
          },
        ],
        "value": 2,
      },
      Object {
        "position": Object {
          "x": 2,
          "y": 1,
        },
        "tiles": Array [
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 1,
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 1,
            },
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 1,
            },
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 1,
            },
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 1,
            },
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 1,
            },
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "value": 1,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 1,
            },
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 1,
            },
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 1,
            },
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "value": 2,
          },
        ],
        "value": 1,
      },
      Object {
        "position": Object {
          "x": 2,
          "y": 2,
        },
        "tiles": Array [
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 2,
            },
            "position": Object {
              "x": 0,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 2,
            },
            "position": Object {
              "x": 0,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 2,
            },
            "position": Object {
              "x": 0,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 2,
            },
            "position": Object {
              "x": 1,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 2,
            },
            "position": Object {
              "x": 1,
              "y": 1,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 2,
            },
            "position": Object {
              "x": 1,
              "y": 2,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 2,
            },
            "position": Object {
              "x": 2,
              "y": 0,
            },
            "value": 2,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 2,
            },
            "position": Object {
              "x": 2,
              "y": 1,
            },
            "value": 0,
          },
          Object {
            "boardPosition": Object {
              "x": 2,
              "y": 2,
            },
            "position": Object {
              "x": 2,
              "y": 2,
            },
            "value": 2,
          },
        ],
        "value": 2,
      },
    ]
  }
  currentPlayer={0}
/>
`;
